uploadNeuroDB/minc_deletion.pl

```diff
@@ -0,0 +1,282 @@
+my @candid = split("/", $dir);
+
+# Let's make directories
+make_path($data_dir . "/archive/"    . $dir) unless(-d  $data_dir . "/archive/"     . $dir);
+make_path($data_dir . "/archive/pic/" . $candid[1]) unless(-d  $data_dir . "/archive/pic/" . $candid[1]);
+make_path($data_dir . "/archive/jiv/" . $candid[1]) unless(-d  $data_dir . "/archive/jiv/" . $candid[1]);
+
+if ($ARGV[0] eq "confirm") {
+  rename($data_dir . "/" . $f->{'File'}, $data_dir . "/archive/" . $f->{'File'});
+  rename($data_dir . "/pic/" . $f->{'VALUE'}, $data_dir . "/archive/pic/" . $f->{'VALUE'});
+  rename($data_dir . "/jiv/" . $jiv_header, $data_dir . "/archive/jiv/" . $jiv_header);
+  rename($data_dir . "/jiv/" . $jiv_raw_byte, $data_dir . "/archive/jiv/" . $jiv_raw_byte);
+}
+
+print "Moving these files to archive:\n";
+print $data_dir . "/" . $f->{'File'} . "\n";
+print $data_dir . "/pic/" . $f->{'VALUE'} . "\n";
+print $data_dir . "/jiv/" . $jiv_header . "\n";
+print $data_dir . "/jiv/" . $jiv_raw_byte . "\n";
+
+
+# Delete from DB
+selORdel("parameter_file","Value");
+selORdel("files_qcstatus","QCStatus");
+selORdel("feedback_mri_comments","Comment");
+# selORdel("mri_protocol_violated_scans","ID");  # if there is data here, the mnc will be in the trashbin
+# selORdel("MRICandidateErrors","Reason");       # not applicable to /assembly
+# selORdel("mri_violations_log","LogID");        # "
+selORdel("files","File");
+
+### Removal of entry in mri_acquisition_dates table ###
+### (if only one file exists and is being removed,  ###
+### the table entry needs to be removed)            ###
+print "\nTarchiveID: $tarchiveid\n";
+print "\nSessionID: $sessionid\n";
+
+# Check #1 in files, if other files from same session
+$query = "select * from files as g " .
+"where g.SessionID=(select f.SessionID from files as f where f.SeriesUID=?) " .
+"and g.SeriesUID <> ?";
+
+$sth = $dbh->prepare($query);
+$rvl = $sth->execute($seriesuid, $seriesuid);
+
+if ($sth->rows > 0) {
+    $sessionfilesfound = 1;
+    print "\nfiles found in the same session\n";
+}
+
+
+# Check #2 in mri_protocol_violated_scans
+$query = "select * from mri_protocol_violated_scans as m " .
+"WHERE m.SeriesUID in (SELECT t.SeriesUID FROM tarchive_series as t " .
+"WHERE t.TarchiveID=?)";
+
+$sth = $dbh->prepare($query);
+$rvl = $sth->execute($tarchiveid);
+
+if ($sth->rows > 0) {
+    $sessionfilesfound = 1;
+    print "\nfiles found in mri_protocol_violated_scans\n";
+}
+
+
+# Check #3 in MRICandidateErrors
+$query = "SELECT * FROM MRICandidateErrors as m " .
+"WHERE m.TarchiveID=?";
+
+$sth = $dbh->prepare($query);
+$rvl = $sth->execute($tarchiveid);
+
+if ($sth->rows > 0) {
+    $sessionfilesfound = 1;
+    print "\nfiles found in MRICandidateErrors\n";
+}
+
+
+# Check #4 in mri_violations_log
+$query = "SELECT * FROM mri_violations_log as m " .
+"WHERE m.TarchiveID=?";
+
+$sth = $dbh->prepare($query);
+$rvl = $sth->execute($tarchiveid);
+
+if ($sth->rows > 0) {
+    $sessionfilesfound = 1;
```

Comment:
You need to define $sessionfilesfound before the if statement. Otherwise, the variable does not exist outside the if statement and it tries to delete the date in mri_acquisition_dates even though there are still files in the files table for that session.